httpcore/_async/http2.py

```diff
@@ -143,10 +143,14 @@ async def handle_async_request(self, request: Request) -> Response:
                 # a GOAWAY event. Other flows of control may then raise
                 # a protocol error at any point they interact with the 'h2_state'.
                 #
-                # In this case we'll have stored the event, and should raise
-                # it as a RemoteProtocolError.
+                # In this case we'll have stored the event, and should raise it.
                 if self._connection_error_event:
-                    raise RemoteProtocolError(self._connection_error_event)
+                    if isinstance(
+                        self._connection_error_event, h2.events.ConnectionTerminated
+                    ):
+                        raise ConnectionNotAvailable(self._connection_error_event)
+                    else:
+                        raise RemoteProtocolError(self._connection_error_event)
```

Comment:
I don't love this repeated `isinstance` check. We could move it into a utility or just always treat `_connection_error_event` as `ConnectionNotAvailable` instead of a protocol error.